File: Engine.Tests.Data/Output/HTML/Syntax Highlighting/C#/Numbers - Input.cs

/*
	Topic: Integers

		(code)

			int x = 12;

			int x = -12;

			long x = 123456789l;

			long x = -123456789L;

			uint x = 12U;

			ulong x = 123456789ul;

		(end)


	Topic: Floating Point

		(code)

			float x = 12;

			float x = -12f;

			float x = 12e1;

			float x = -12E-1F;

			double x = .012;

			double x = -.012D;

			double x = .012E+5d;

			double x = -.012e10;

			decimal x = 0.0123M;

			decimal x = -123.123;

			decimal x = 0.0123E10;

			decimal x = -123.123e-9m;

		(end)


	Topic: Hex and Binary

		(code)

			int x = 0x01230ABC;

			ulong x = 0xabc12345UL;

			int x = 0b00100110;

			ulong x = 0B0110100110011100ul;

		(end)


	Topic: Digit Separators

		(code)

			int x = 1_234_567;

			ulong x = 0xabc1_2345UL;

			ulong x = 0B0110_1001_1001_1100ul;

			float x = 1.234_567e1_2;

			int x = 1____2____3;

		(end)


	Topic: Traps

		The fact that the hex value ends in E shouldn't make it think the following token is part of
		the number as an exponent.

		(code)

			int x = 0x000E+a;

			int x = 0x000e-1;

		(end)


		The minus sign should be part of the number for -1 but not for x-1.

		(code)

			int x = -1;

			int x = y-1;

			int x = y -1;

			int x = 1-1;

			int x = 1 -1;

			int x = (-1);

			int x = 1 + -1;

		(end)


		Digit separators can't start a number.

		(code)

			int x = _12;

		(end)
*/

File: Tests.Data/Output/HTML/Syntax Highlighting/Kotlin/Numbers - Input.kt

/*
	Topic: Integers

		--- Code
		12
		-12
		123456789l
		-123456789L
		12U
		123456789ul
		---


	Topic: Floating Point

		--- Code
		12e1
		-12E-1
		12.34f
		-12.34F
		12.34e+5
		-12.34e-5f
		.012
		-.012F
		.012E+5
		-.012e10
		0.0123f
		-0.0123F
		0.0123e+5
		-0.0123E-5
		---


	Topic: Hex and Binary

		--- Code
		0x01230ABC
		0xabc12345UL
		0b00100110
		0B0110100110011100ul
		---


	Topic: Digit Separators

		--- Code
		1_234_567
		0xabc1_2345UL
		0x_ab12_cd45UL
		0B0110_1001_1001_1100ul
		0b_0110_1001_1001_1100ul
		1.234_567e1_2
		1____2____3
		---


	Topic: Traps

		The fact that the hex value ends in E shouldn't make it think the following token is part of
		the number as an exponent.

		--- Code
		0x000E+a
		0x000e-1
		---

		The minus sign should be part of the number for -1 but not for x-1.

		--- Code
		-1
		y-1
		y -1
		1-1
		1 -1
		(-1)
		1 + -1
		---

		Digit separators can't start a number.

		--- Code
		_12
		---

		Numbers appearing in other contexts shouldn't be mistaken for constants.

		--- Code
		a1_2_3 = "45"
		---
*/
